draft-ietf-quic-transport.md

```diff
@@ -542,68 +545,132 @@ the server supports, as shown below.
-* Octets 13-16: Version
-* Octets 17+: Payload
+### Client Initial Packet {#packet-client-initial}
 
-The client MUST choose a random 64-bit value and use it as the initial
-Connection ID in all packets until the server replies with the final Connection
-ID. The server echoes the client's Connection ID in Non-Final Server Cleartext
-packets.  The first Final Server Cleartext and all subsequent packets MUST use
-the final Connection ID, as described in {{connection-id}}.
+The Client Initial packet has a type octet of 0x82.  It carries the first
+cryptographic handshake message sent by the client, prior to receiving any
+packets from the server.
 
-The payload of a Cleartext packet consists of a sequence of frames, as described
-in {{frames}}.  Cleartext packets also include an integrity check, described in
-{{QUIC-TLS}}.
+The client populates the connection ID and packet number fields with randomly
+selected values.  The randomized packet number is incremented for each
+subsequent packet (see {{packet-numbers}}).
+
+The payload of a Client Initial packet consists of a STREAM frame (or frames)
+for stream 0 containing a cryptographic handshake message, plus PADDING frames
+that increase the size of the packet to a minimum size (see {{packetization}}).
+
+The client uses the Client Initial Packet type for any packet that contains an
+initial cryptographic handshake message.  This includes all cases where a new
+packet containing the initial cryptographic message needs to be created, either
+in response to a Version Negotiation packet, in response to a stateless retry
+({{packet-server-stateless}}).
+
+
+### Server Stateless Retry Packet {#packet-server-stateless}
+
+A Server Stateless Retry packet has a type octet of 0x83.  It carries
+cryptographic handshake messages and acknowledgments.  It is used by a server
+that wishes to perform a stateless retry (see {{stateless-retry}}).
+
+The connection ID field in a Server Stateless Retry packet contains a server
+selected value, see {{connection-id}}.
+
+The packet number contains the packet number from the packet that caused it to
+be sent.  This allows a client to verify that the server received its packet.
+
+After receiving a Server Stateless Retry packet, the next cryptographic
+handshake message MUST be sent by the client on stream 0 at an offset of 0.  The
+client discards any buffers it uses for generating retransmissions of handshake
+data and effectively starts the connection anew, keeping only the cryptographic
+handshake state.  The cryptographic handshake is continued, but the transport
+connection is reset.  Continuing the cryptographic handshake is necessary to
+ensure that an attacker cannot force a downgrade of any cryptographic
+parameters.
+
+The client constructs a new Client Cleartext packet
+({{packet-client-cleartext}}) for subsequent cryptographic handshake messages.
+
+The payload of this packet contains STREAM frames and could contain PADDING and
+ACK frames.
```

Comment:
Swap the order of these last two paragraphs, since "this packet" is ambiguous in this last para, and since the note about Client Cleartext packet fits just before the next section.